src/Microsoft.Azure.NotificationHubs/NotificationHubClient.cs

```diff
@@ -2186,7 +2156,21 @@ private void SetUserAgent()
                 xmlReader.ReadToDescendant("content");
                 xmlReader.ReadStartElement();
 
-                return (TEntity)_entitySerializer.Deserialize(xmlReader, xmlReader.Name);
+                var entity = _entitySerializer.Deserialize(xmlReader, xmlReader.Name);
+
+                if (entity is GcmTemplateRegistrationDescription)
```

Comment:
No need to as casting later if you do `entity is GcmTemplateRegistrationDescription gcmTemplateRegistration`

Reply:
Specifically using this C# 7 syntax:
https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/is#type-pattern

Reply:
have fixed